pennylane/qchem/integrals.py

```diff
@@ -274,12 +280,16 @@ def gaussian_overlap(la, lb, ra, rb, alpha, beta):
     return s
 
 
-def overlap_integral(basis_a, basis_b, normalize=True):
+def overlap_integral(basis_a, basis_b, argnum=None, normalize=True):
```

Comment:
I was trying to use this as given below:
```py
import pennylane as qml
from pennylane import numpy as np
import matplotlib.pyplot as plt

symbols = ["H", "H"]
geometry = np.array([[-0.672943567415407, 0.0, 0.0],
                     [ 0.672943567415407, 0.0, 0.0]], requires_grad=True)

mol = qml.qchem.Molecule(symbols, geometry, argnum=[0])
S1, S2 = mol.basis_set[0], mol.basis_set[1]
```
```py
>>> qml.grad(qml.qchem.overlap_integral(S1, S2))(geometry)
array([[ 0.39219112,  0.        ,  0.        ],
       [-0.39219112,  0.        ,  0.        ]])
```
but `jax` equivalent of the above code doesn't work:

```py
import pennylane as qml
import jax
from jax import numpy as np

jax.config.update("jax_enable_x64", True)

symbols = ["H", "H"]
geometry = np.array([[-0.672943567415407, 0.0, 0.0],
                     [ 0.672943567415407, 0.0, 0.0]], dtype=np.float64)

mol = qml.qchem.Molecule(symbols, geometry, argnum=[0])
S1, S2 = mol.basis_set[0], mol.basis_set[1]
```

```py
>>> jax.grad(qml.qchem.overlap_integral(S1, S2))(geometry)
Array([[0., 0., 0.],
       [0., 0., 0.]], dtype=float64)
```

Reply:
Removing the grad computation here and checking the results it seems like they're equivalent. Not sure why using jax.grad vs qml.grad gives you different values...

Reply:
As in doing
`qml.qchem.overlap_integrals(S1, S2)(geometry)`
with either jax or pnp gives the same value (just one is a tensor the other is not).

Reply:
Yep non-gradient computation is correct but I am more concerned about us breaking the `grad` calculation here with `jax`.

Reply:
Question: since `overlap_integrals` doesn't actually use geometry, how come you can get a gradient based on the differentiability of the geometry? Is there something that's changing the basis functions based on the geometry when we do `qml.grad(qml.qchem.overlap_integrals(S1, S2))(geometry)`? I'm just confused about gradients at this point 😅

Reply:
fixed so i resolve 😎